Orange/widgets/data/owgroupby.py

```diff
@@ -0,0 +1,490 @@
+                return QItemSelectionModel.NoUpdate
+        if (
+            event is not None
+            and event.type() == QEvent.MouseMove
+            and flags & QItemSelectionModel.ToggleCurrent
+        ):
+            # Disable ctrl drag 'toggle'; can be made to deselect the last
+            # index, would need to keep track of the current selection
+            # (selectionModel does this but does not expose it)
+            flags &= ~QItemSelectionModel.Toggle
+            flags |= QItemSelectionModel.Select
+        return flags
+
+
+class CheckBox(QCheckBox):
+    def __init__(self, text, parent):
+        super().__init__(text)
+        self.parent: OWGroupBy = parent
+
+    def nextCheckState(self) -> None:
+        """
+        Custom behaviour for switching between steps. It is required since
+        sometimes user will select different types of attributes at the same
+        time. In this case we step between unchecked, partially checked and
+        checked or just between unchecked and checked - depending on situation
+        """
+        if self.checkState() == Qt.Checked:
+            # if checked always uncheck
+            self.setCheckState(Qt.Unchecked)
+        else:
+            agg = self.text()
+            selected_attrs = self.parent.get_selected_attributes()
+            types_ = set(type(attr) for attr in selected_attrs)
+            can_be_applied_all = (AGGREGATIONS[agg].types & types_) == types_
+
+            # true if aggregation applied to all attributes that can be
+            # aggregated with selected aggregation
+            applied_all = all(
+                type(attr) not in AGGREGATIONS[agg].types
+                or agg in self.parent.aggregations[attr]
+                for attr in selected_attrs
+            )
+            if self.checkState() == Qt.PartiallyChecked:
+                # if partially check: 1) check if agg can be applied to all
+                # 2) else uncheck if agg already applied to all
+                # 3) else leve partially checked to apply to all that can be aggregated
+                if can_be_applied_all:
+                    self.setCheckState(Qt.Checked)
+                elif applied_all:
+                    self.setCheckState(Qt.Unchecked)
+                else:
+                    self.setCheckState(Qt.PartiallyChecked)
+                    # since checkbox state stay same signal is not emitted
+                    # automatically but we need a callback call so we emit it
+                    self.stateChanged.emit(Qt.PartiallyChecked)
+            else:  # self.checkState() == Qt.Unchecked
+                # if unchecked: check if all can be checked else partially check
+                self.setCheckState(
+                    Qt.Checked if can_be_applied_all else Qt.PartiallyChecked
+                )
+
+
+@contextmanager
+def block_signals(widget):
+    widget.blockSignals(True)
+    yield
+    widget.blockSignals(False)
```

Comment:
I think you must do

```
try:
    yield
finally:
    widget.blockSignals(False)
```

(https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager)